pgjdbc/src/main/java/org/postgresql/ssl/LibPQFactory.java

```diff
@@ -122,14 +128,36 @@ public LibPQFactory(Properties info) throws PSQLException {
         }
         try {
           CertificateFactory cf = CertificateFactory.getInstance("X.509");
-          // Certificate[] certs = cf.generateCertificates(fis).toArray(new Certificate[]{}); //Does
-          // not work in java 1.4
+
           Object[] certs = cf.generateCertificates(fis).toArray(new Certificate[]{});
           ks.load(null, null);
           for (int i = 0; i < certs.length; i++) {
             ks.setCertificateEntry("cert" + i, (Certificate) certs[i]);
           }
+
           tmf.init(ks);
+
+          String sslcrlfile =  PGProperty.SSL_CRL_FILE.get(info);
+          if (sslcrlfile == null ) {
+            sslcrlfile = defaultdir + "root.crl";
+          }
+
+          X509CRL crl = loadRevokedCertificates(cf, sslcrlfile);
+
+          if ( crl != null  ) {
+
+            for (X509Certificate cert:km.getCertificateChain(null) ) {
+              X509CRLEntry e = crl.getRevokedCertificate(cert.getSerialNumber());
+
+              if ( e != null && (e.getRevocationDate().compareTo( Calendar.getInstance().getTime()) < 0) ) {
```

Comment:
You could use the [before](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#before-java.util.Date-) method on `Date` rather than comparing return of compare to 0.